test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -5975,13 +6012,76 @@ def make_input(batch_size):
 
     def _gru_test(self, layers, bidirectional, initial_state,
                   packed_sequence, dropout):
+
+        class GRUWithStateModel(torch.nn.Module):
+            def __init__(self, layers, bidirect, dropout, batch_first):
+                super(GRUWithStateModel, self).__init__()
+
+                self.batch_first = batch_first
+                self.inner_model = torch.nn.GRU(RNN_INPUT_SIZE, RNN_HIDDEN_SIZE, num_layers=layers, 
+                                                bidirectional=bidirectional, dropout=dropout,
+                                                batch_first=batch_first)
+
+            def forward(self, input: PackedSequence, hx):
+                return self.inner_model(input, hx)
+
+        class GRUWithoutStateModel(torch.nn.Module):
+            def __init__(self, layers, bidirect, dropout, batch_first):
+                super(GRUWithoutStateModel, self).__init__()
+                self.batch_first = batch_first
+                self.inner_model = torch.nn.GRU(RNN_INPUT_SIZE, RNN_HIDDEN_SIZE, num_layers=layers, 
+                                                bidirectional=bidirectional, dropout=dropout,
+                                                batch_first=batch_first)
+
+            def forward(self, input: PackedSequence):
+                return self.inner_model(input)
+
+        class GRUNoSeqLengthWithoutStateModel(torch.nn.Module):
+            def __init__(self, layers, bidirect, dropout, batch_first):
+                super(GRUNoSeqLengthWithoutStateModel, self).__init__()
+                self.batch_first = batch_first
+                self.inner_model = torch.nn.GRU(RNN_INPUT_SIZE, RNN_HIDDEN_SIZE, num_layers=layers, 
+                                                bidirectional=bidirectional, dropout=dropout,
+                                                batch_first=batch_first)
+
+            def forward(self, input):
+                return self.inner_model(input)
+
+        class GRUNoSeqLengthWithStateModel(torch.nn.Module):
+            def __init__(self, layers, bidirect, dropout, batch_first):
+                super(GRUNoSeqLengthWithStateModel, self).__init__()
+                self.batch_first = batch_first
+                self.inner_model = torch.nn.GRU(RNN_INPUT_SIZE, RNN_HIDDEN_SIZE, num_layers=layers, 
+                                                bidirectional=bidirectional, dropout=dropout,
+                                                batch_first=batch_first)
+
+            def forward(self, input, hx):
```

Comment:
Do we need special annotations for input here? Or is it a regular Tensor?

Reply:
It's just a regular Tensor which would be inferred by Python, so we don't need an annotation.